django/core/files/storage.py

```diff
@@ -140,24 +143,98 @@ def url(self, name):
         raise NotImplementedError('subclasses of Storage must provide a created_time() method')
 
     def modified_time(self, name):
         """
         Returns the last modified time (as datetime object) of the file
-        specified by name.
+        specified by name. Deprecated: use get_modified_time() instead.
         """
+        warnings.warn(
+            'Storage.modified_time() is deprecated in favor of get_modified_time().',
+            RemovedInDjango20Warning,
+            stacklevel=2,
+        )
         raise NotImplementedError('subclasses of Storage must provide a modified_time() method')
 
+    def get_accessed_time(self, name):
+        """
+        Return the last accessed time (as a datetime object) of the file
+        specified by name. The datetime will be timezone-aware if USE_TZ=True.
+        """
+        # At the end of the deprecation:
+        # raise NotImplementedError('subclasses of Storage must provide a get_accessed_time() method')
+        warnings.warn(
+            'Storage.accessed_time() is deprecated. '
+            'Storage backends should implement get_accessed_time().',
+            RemovedInDjango20Warning,
+            stacklevel=2,
+        )
+        dt = self.accessed_time(name)
+        return _possibly_make_aware(dt)
+
+    def get_created_time(self, name):
+        """
+        Return the creation time (as a datetime object) of the file
+        specified by name. The datetime will be timezone-aware if USE_TZ=True.
+        """
+        # At the end of the deprecation:
+        # raise NotImplementedError('subclasses of Storage must provide a get_created_time() method')
+        warnings.warn(
+            'Storage.created_time() is deprecated. '
+            'Storage backends should implement get_created_time().',
+            RemovedInDjango20Warning,
+            stacklevel=2,
+        )
+        dt = self.created_time(name)
+        return _possibly_make_aware(dt)
+
+    def get_modified_time(self, name):
+        """
+        Return the last modified time (as a datetime object) of the file
+        specified by name. The datetime will be timezone-aware if USE_TZ=True.
+        """
+        # At the end of the deprecation:
+        # raise NotImplementedError('subclasses of Storage must provide a get_modified_time() method')
+        warnings.warn(
+            'Storage.modified_time() is deprecated. '
+            'Storage backends should implement get_modified_time().',
+            RemovedInDjango20Warning,
+            stacklevel=2,
+        )
+        dt = self.modified_time(name)
+        return _possibly_make_aware(dt)
+
+
+def _possibly_make_aware(dt):
+    """
+    Convert a datetime object in the local timezone to aware
+    if USE_TZ is True.
+    """
+    # This function is only needed to help with the deprecations above and can
+    # be removed in Django 2.0, RemovedInDjango20Warning.
+    tz = timezone.get_default_timezone()
+    return timezone.make_aware(dt, tz)
```

Comment:
Great, looks like `.astimezone(timezone.utc)` on Windows doesn't actually work anyway. (Although possibly it's more subtle than that, and the test isn't setting all the things that Windows needs to set the local time correctly. That it passed all the others is suspicious, though.)